clients/src/test/java/org/apache/kafka/common/serialization/SerializationTest.java

```diff
@@ -107,6 +103,23 @@ public void stringSerdeShouldSupportDifferentEncodings() {
         }
     }
 
+    @Test
```

Comment:
I would add test for all primitive types. The test should also check the expected `byte[]` array size after serialization and test a "round trip".

We should also have a test for non-primitive type round-trip.

Lastly, I would add a test for deserializing different list-types.

Also `null` corer case should be tested.

Reply:
@mjsax Where should I place all of these new test cases? Should I create a new class?

Reply:
Add new test methods to this test should be sufficient.

Reply:
Ok I added the following tests:
`listSerdeShouldRoundtripPrimitiveInput(): Arrays.asList(1, 2, 3)`
`listSerdeShouldRountripNonPrimitiveInput(): Arrays.asList("A", "B", "C")`
`listSerdeShouldReturnEmptyCollection(): Arrays.asList()`
`listSerdeShouldReturnNull(): null`
`listSerdeSerializerShouldReturnByteArrayOfSize(): Arrays.asList(1, 2, 3) => 16`
`listSerdeShouldReturnLinkedList() new LinkedList<>()`
`listSerdeShouldReturnStack() new Stack<>()`

I think I covered it all. Btw you said *all primitive types*, you mean all 6 of them, right?

Reply:
These new tests look good. I agree, though, that we should test round trip + length for each of the primitive types (short, integer, long, float, double, and UUID). Just because it would be easy to mess up just one of them, so we really should have test coverage for them all.